tests/Aspire.TestUtilities/RequiresDockerAttribute.cs

```diff
@@ -29,4 +28,14 @@ public RequiresDockerAttribute(string? reason = null)
     {
         Reason = reason;
     }
+
+    public IReadOnlyCollection<KeyValuePair<string, string>> GetTraits()
+    {
+        if (!IsSupported)
+        {
+            return [new KeyValuePair<string, string>(XunitConstants.Category, "failing")];
+        }
+
+        return [];
+    }
```

Comment:
So, there are no more discoverers, which is cool.

How does this work though? Where does `IsSupported` come from and how is it set?

Reply:
IsSupported was, and is still, defined in the attribute class

https://github.com/dotnet/aspire/blob/ad75817e69a1494b6c878ece595b0dfc42801e5f/tests/Aspire.TestUtilities/RequiresDockerAttribute.cs#L23-L26

Just that traits calculation is moved from a special trait discoverer to the attribute class itself.